### PR TITLE
Rotary volume logic for both high res and low res devices

### PR DESCRIPTION
**Pull request by laiyichin**
#### WHAT
Add support for high / low resolution rotary devices to make volume changes.

#### WHY
So that sensitivity for different types of devices can tune volume with their rotary with good user experiences.

#### HOW
- Modify the rotary accumulator to count low res devices positive change as 1 and negative change as -1
- Add modifier for RotaryVolumeControls, and take the changes as is for low res devices and do a 1 pixel = 0.1% volume change for high res devices.

#### Checklist :clipboard:
- [X] Add explicit visibility modifier and explicit return types for public declarations
- [X] Run spotless check
- [X] Run tests
- [X] Update metalava's signature text files

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/RotaryInputAccumulator.kt [hunk 1]
```diff
@@ -38,16 +39,27 @@ internal class RotaryInputAccumulator(
     public fun onRotaryScroll(scrollPixels: Float, eventTimeMillis: Long) {
         val timeSinceLastAccumulatedMs = eventTimeMillis - lastAccumulatedEventTimeMs
         lastAccumulatedEventTimeMs = eventTimeMillis
+
+        val change = if (isLowRes && scrollPixels > 0f) { // For positive tick in low res devices
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/RotaryInputAccumulator.kt:43]: with these first three branches, and then the different handling in onEventAccumulated, it feels like you might really want to simplfy this into two major cases

if (lowRes) {
 doImmediately
} else {
  current logic
}

**Comment by yschimke** [comment]
@Kpeved want to review?

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/RotaryInputAccumulator.kt (hunk 1 above)
Comment (laiyichin) [reply compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/RotaryInputAccumulator.kt:43]: I have considered this, and tested it feeling like if we don't accumulate lowRes the volume change is still quite slow. I can check with UX again on this to verify what is a better experience.

**Comment by yschimke** [comment]
I'll do a fuller review on Monday, but I feel out of my depth.

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/RotaryInputAccumulator.kt (hunk 1 above)
Comment (laiyichin) [reply compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/RotaryInputAccumulator.kt:43]: This `onRotaryScroll` method is accumulating the changes when `timeSinceLastAccumulatedMs > eventAccumulationThresholdMs`; for high res devices, it'd be accumulating the pixels,  and for low res devices, it'd be accumulating by notches.

These changes then are passed into the `RotaryVolumeControls`'s `onRotaryInputAccumulated`, high res accumulated pixels are converted into volume (by the 1 pixel == 0.1% volume ratio), and low res notches are converted directly with the ratio of 1 notch : 1 volume change.

File: media/audio-ui/src/main/java/com/google/android/horologist/audio/ui/VolumeScreen.kt [hunk 2]
```diff
@@ -223,3 +228,8 @@ private fun volumeDescription(volumeUiState: VolumeUiState, isAudioOutputConnect
         stringResource(id = R.string.horologist_volume_screen_not_connected_state)
     }
 }
+
+@ExperimentalHorologistApi
+@Composable
+private fun isLowResInput(): Boolean = LocalContext.current.packageManager
```
[the remark this answers is not in the document]
Comment (laiyichin) [reply media/audio-ui/src/main/java/com/google/android/horologist/audio/ui/VolumeScreen.kt:234]: Yeah I took this from https://github.com/google/horologist/blob/49f98802cbcff8ad2cbab595d6d0c8317ce16b11/compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Rotary.kt#L328-L331

Still thinking where could be a good place for this. Suggestion?

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/AccumulatedRotaryInputModifier.kt [hunk 3]
```diff
@@ -42,10 +42,11 @@ import com.google.android.horologist.annotations.ExperimentalHorologistApi
 @ExperimentalHorologistApi
 public fun Modifier.onRotaryInputAccumulatedWithFocus(
     focusRequester: FocusRequester? = null,
+    isLowRes: Boolean = false,
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/AccumulatedRotaryInputModifier.kt:45]: Can we default this to the right value?

**Review by yschimke — APPROVED** [review]

**Pull request merged**